[gongg21] iP

gongg21 opened the pull request:
### GBOT

> It's so good - User 41312

GBot is a simple **task-tracking chatbot** that is:
- Easy to use
- Easy to learn
- Versatile

How to use?
1. Start the chatbot.
2. Type in a keyword (and appropriate follow-up message)
3. Hit enter
4. Done 😃

**Keywords** to use:
1. `list`: to list your existing tasks
2. `todo (task)`: to add a Todo task
3. `deadline (task) /by (deadline)`: to add a Deadline task
4. `event (task) /from (start) /to (end)`: to add an Event task
5. `delete (task number)`: to delete a task
6. `mark (task number)`: to mark a task as done
7. `unmark (task number)`: to unmark a task
8. `find (keyword)`: relevant tasks using a keyword

Click [here](https://github.com/gongg21/ip) to view the codebase!

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```hs
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Features (to be implemented):
- [ ] Checkstyle
- [ ] GUI
- [ ] Varargs

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

gongg21 committed:
no message

gongg21 committed:
Level-1

gongg21 committed:
Level-2

gongg21 committed:
Level-3

gongg21 committed:
Level-4

gongg21 committed:
A-TextUITesting

gongg21 committed:
Level-5

gongg21 committed:
Level-6

gongg21 committed:
A-Enums

gongg21 committed:
Add Level-7 files

gongg21 committed:
Add Level-7 files

gongg21 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Level-7 files

gongg21 committed:
Add Level-7 Files

gongg21 committed:
Revert "Add Level-7 files"

This reverts commit ceb90635a079c71d4229853017263c773b918da7.

gongg21 committed:
Revert "Add Level-7 files"

This reverts commit 751f611461e96418913159f9922f2c7e32d880ba.

gongg21 committed:
Revert "Add Level-7 files"

This reverts commit 751f611461e96418913159f9922f2c7e32d880ba.

gongg21 committed:
Revert "Revert "Add Level-7 files""

This reverts commit 718aeaa4a333782a22c82128692be79857010299.

gongg21 committed:
Revert "Revert "Revert "Add Level-7 files"""

This reverts commit 2a854f62fbd9c8ba48c85659f604252ac787b64e.

gongg21 committed:
Add Level-8 Files

gongg21 committed:
Add Parser.java and Ui.java

gongg21 committed:
Move task files to Tasks package

gongg21 committed:
Merge branch 'master' into 'add-gradle-support'

gongg21 committed:
Add JUnit Test files

gongg21 committed:
Update build-gradle file for JAR creation

gongg21 committed:
Add JavaDoc to all main files.

gongg21 committed:
Update Coding Standards for main files

gongg21 committed:
Update TaskList.java and Parser.java for find() method.

gongg21 committed:
Add JavaDocs for find() method

gongg21 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/exceptions/GBotException.java
#	src/main/java/gbot/Ui.java
#	src/main/java/tasks/Event.java
#	src/main/java/tasks/Task.java

gongg21 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/gbot/Ui.java
#	src/main/java/tasks/Task.java

gongg21 committed:
Update files with invalid syntax

gongg21 committed:
Update files for JUnit tests

eyelessrhyme7 commented on src/main/java/exceptions/DeadlineException.java:
```diff
@@ -0,0 +1,16 @@
+    public DeadlineException() {
+        super("Invalid format for Deadline task. Please adhere to the following:\n" +
+                "deadline (task) /by (deadline)");
+    }
```
consider giving example input to guide user better.

eyelessrhyme7 commented on src/main/java/gbot/Parser.java:
```diff
@@ -0,0 +1,117 @@
+        }
+
+        String prefix = message.split(" ")[0];
+        String str = message.substring(prefix.toString().length() + 1);
```
Consider a different name for str, something more descriptive for clearer understanding.

eyelessrhyme7 commented on src/main/java/gbot/Storage.java:
```diff
@@ -0,0 +1,109 @@
+     *
+     * @param list The list of tasks to update the file.
+     */
+    public void updateFile(ArrayList<Task> list) {
```
Consider renaming this function to writeToFile or something different than update, as update can be interpreted as adding/appending too.

eyelessrhyme7 commented on src/main/java/gbot/StorageStub.java:
```diff
@@ -0,0 +1,49 @@
+     * @return An empty ArrayList.
+     */
+    @Override
+    public ArrayList<Task> load() {
```
Consider renaming function name to be more descriptive.

eyelessrhyme7 commented on src/main/java/gbot/TaskList.java:
```diff
@@ -0,0 +1,180 @@
+    private Storage storage;
+    private ArrayList<Task> list;
+    private int taskCount;
+    private static final String LINE = "____________________________________________________________";
```
Consider deleting LINE or placing it in a different class, maybe under Ui

eyelessrhyme7 commented on src/main/java/gbot/Ui.java:
```diff
@@ -0,0 +1,44 @@
+     * Prints the initial greeting to user.
+     */
+    public void ask() {
+        print("Hello I'm GBot!\nWhat can I do for you?");
```
Consider placing the entryGreeting and exitGreeting as constants rather than inside a function. Use functions only when a true verb/action is being performed

eyelessrhyme7 commented on src/main/java/gbot/Ui.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Prints the closing message to user.
+     */
+    public void end() {
```
Consider renaming function to a verb and be more descriptive, such as, displayExitMessage(). Although, if a function only returns a string, consider placing string as a static attribute or constant

eyelessrhyme7 reviewed (commented):
Good use of abstractions and OOP. Keeps the Gbot class clean and easy to read. Recommend minor naming changes and replacing functions with strings. (Eg, ask(), end() in Ui)

feifeiraindrops commented on src/main/java/gbot/Parser.java:
```diff
@@ -0,0 +1,117 @@
+        switch (prefix) {
+        case "mark":
+            if (str.isBlank()) {
+                return "Please enter a task number.";
```
You can consider saving the strings that are reused as constant

feifeiraindrops commented on src/main/java/gbot/TaskList.java:
```diff
@@ -0,0 +1,180 @@
+        taskCount++;
+
+        String count = "\nNow you have " + taskCount + " tasks to do.";
+        return "Got it. I've added this task:\n" + newEvent + count;
```
You can consider putting these interactions under the Ui class and have different methods in the Ui class to handle interactions given different commands

feifeiraindrops commented on src/main/java/gbot/Ui.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Prints the initial greeting to user.
+     */
+    public void ask() {
```
You can consider renaming the name of this method to be more descriptive. "ask" might not be the most accurate description of what this method is doing

feifeiraindrops commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+ * @author Gallen Ong
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
maybe this variable can be named using a noun instead, since "by" can have multiple meanings

feifeiraindrops reviewed (commented):
Very good naming practices!

gongg21 committed:
Add and update files for Level-10

gongg21 committed:
Update README.md

gongg21 committed:
Update build.gradle file

gongg21 committed:
Merge branch 'master' of https://github.com/gongg21/ip

gongg21 committed:
Update Parser.java with assertions

gongg21 committed:
Update files for A-CodeQuality

Adhering to the code quality guidelines, redundant imports were removed,
incorrect indentations were fixed, blank lines were removed, and access
modifiers were adjusted appropriately.

gongg21 committed:
Merge 'master' and 'branch-A-Assertions' branch

Update files for A-Assertions

gongg21 committed:
Merge branch 'master'

gongg21 committed:
Merge branch 'master' into branch-A-CodeQuality

gongg21 committed:
Merge 'master' and 'branch-A-CodeQuality' branch

Update files for A-CodeQuality

gongg21 committed:
Merge branch 'master'

gongg21 committed:
Update files from comments

As per comments received, several files were edited and changed.
Several methods name in Ui.java and Storage.java were changed.
Variable names in Deadline.java and Event.java were changed.

gongg21 committed:
Update files for BCD-Extension

Added methods in Parser.java and TaskList.java for C-MassOps such as
deleteAll, markAll, and unmarkAll.

Added help functionality for C-Help.

gongg21 committed:
Add JavaDocs in files

Added JavaDocs for new methods and cleaned up files.

gongg21 committed:
Update files for A-Personality

Chatbot responses have been updated to exhibit a more posh chatbot.

gongg21 committed:
Update files for A-BetterGUI

Changed images for chatbot and user. Edited the GUI screen slightly.

gongg21 committed:
Merge branch 'branch-A-BetterGUI'

gongg21 committed:
Update README.md and cleaned files

README.md was updated with detailed User Guide for the chatbot.
GUI and files were cleaned up with updated responses.

gongg21 committed:
Update Ui.png

gongg21 committed:
Update build.gradle

Set the mainClass for shawdowJar to be Main.

gongg21 committed:
Update GUI and files for handling errors

TodoException, DeadlineException, and EventException classes were
removed. Handling of errors in several files were modified.

Parser.java was edited significantly to separate methods for parsing.

GUI was improved to display full text.

gongg21 committed:
Update GUI-related files

Ui.png was improved. Credit on reuse for GUI-related files was added.
Minor edits were made to remove redundant lines.

gongg21 committed:
Update files for error handling

Bugs in using deadline and event command were fixed.
Error handling updated in several files.
mainClass in build.gradle was updated to fix JavaFX error.

gongg21 committed:
Update TaskList.java

find method was updated to fix bug.
